src/PluginManagerInterface.php

```diff
@@ -19,11 +19,38 @@ interface PluginManagerInterface extends ServiceLocatorInterface
     /**
      * Validate an instance
      *
-     * @return void
      * @throws InvalidServiceException If created instance does not respect the
      *     constraint on type imposed by the plugin manager.
      * @throws ContainerExceptionInterface If any other error occurs.
      * @psalm-assert InstanceType $instance
      */
-    public function validate(mixed $instance);
+    public function validate(mixed $instance): void;
+
+    public function has(string $id): bool;
+
+    /**
+     * @template TRequestedInstance extends InstanceType
+     * @psalm-param class-string<TRequestedInstance>|string $id Service name of plugin to retrieve.
+     * @psalm-return ($id is class-string ? TRequestedInstance : InstanceType)
+     * @throws Exception\ServiceNotFoundException If the manager does not have
+     *     a service definition for the instance, and the service is not
+     *     auto-invokable.
+     * @throws InvalidServiceException If the plugin created is invalid for the
+     *     plugin context.
+     */
+    public function get(string $id): mixed;
```

Comment:
Yah, `has` is superfluous indeed. 
I guess I've added it to also add some `psalm-assert` on it to provide information of what will be returned from `get`. But since `has` actually does no validation at all, it would provide false information and thus we can safely drop it here.